meson.build

```diff
@@ -91,7 +91,20 @@ endif
 
 # Check for libsystemd availability. Optional, only required for MCTP dbus scan
 libsystemd_dep = dependency('libsystemd', version: '>219', required: false)
-conf.set('CONFIG_LIBSYSTEMD', libsystemd_dep.found(), description: 'Is libsystemd(>219) available?')
+if libsystemd_dep.found()
+  if cc.links('''#include <systemd/sd-bus.h>
+                 int main(void) {
+                     struct sd_bus *ret;
+                     return sd_bus_default(&ret);
+                 }
+               ''',
+              dependencies: libsystemd_dep,
+              name: 'Link check')
+       conf.set('CONFIG_LIBSYSTEMD', libsystemd_dep.found(), description: 'Is libsystemd(>219) available?')
+  else
+    libsystemd_dep = dependency('', required: false)
```

Comment:
meson also provides `declare_dependency()`, which allows declaring a dependency to a missing library.

Reply:
Ah, didn't see that in the documentation. Looks a bit saner. Right let me fix that up.